webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx

```
@@ -0,0 +1,1335 @@
+          if (
+            !supportedChains.find(c => +c.chainId === ChainId.MATIC_MAINNET)
+          ) {
+            supportedChains.push({
+              chainId: ChainId.MATIC_MAINNET.toString(),
+              networkName: 'Polygon',
+              nativeCurrency: {
+                name: 'MATIC',
+                symbol: 'MATIC',
+                decimals: 18,
+                icon:
+                  'https://raw.githubusercontent.com/0xsquid/assets/main/images/tokens/matic.svg'
+              }
+            } as ChainData)
+          }
+          setProviderChains(
+            supportedChains.filter(c =>
+              CROSS_CHAIN_SUPPORTED_CHAINS.includes(+c.chainId)
+            )
+          )
+          setProviderTokens(
+            supportedTokens.filter(t =>
+              CROSS_CHAIN_SUPPORTED_CHAINS.includes(+t.chainId)
+            )
+          )
+        }
+      } catch (error) {
+        console.log('error: ', error)
+      }
+    })()
+  }, [crossChainProvider, wallet])
+
+  // calculates Route for the selectedToken
+  const calculateRoute = useCallback(async () => {
+    const abortController = abortControllerRef.current
+    const signal = abortController.signal
+
+    const providerMANA = providerTokens.find(
+      t =>
+        t.address.toLocaleLowerCase() === destinyChainMANA.toLocaleLowerCase()
+    )
+    if (
+      !crossChainProvider ||
+      !crossChainProvider.isLibInitialized() ||
+      !wallet ||
+      !selectedToken ||
+      !providerMANA
+    ) {
+      return
+    }
+    try {
+      setRoute(undefined)
+      setIsFetchingRoute(true)
+      setRouteFailed(false)
+      let route: RouteResponse | undefined = undefined
+      const fromAmountParams = {
+        fromToken: selectedToken,
+        toAmount: ethers.utils.formatEther(
+          order
+            ? order.price
+            : !isNFT(asset) && +asset.price > 0
+            ? asset.price
+            : 1 //TODO: review this
```

should we raise an error/alert if there's no price for the asset?